testsuite/src/test/java/org/ironjacamar/core/workmanager/BootstrapContextTestCase.java

```diff
@@ -0,0 +1,130 @@
+import org.ironjacamar.rars.ResourceAdapterFactory;
+import org.ironjacamar.rars.wm.WorkConnection;
+import org.ironjacamar.rars.wm.WorkConnectionFactory;
+
+import javax.annotation.Resource;
+
+import org.jboss.shrinkwrap.api.spec.ResourceAdapterArchive;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Basic WorkManager test case
+ * @author <a href="mailto:jesper.pedersen@ironjacamar.org">Jesper Pedersen</a>
+ */
+@RunWith(IronJacamar.class)
+@Configuration(full = true)
+@PreCondition(condition = AllChecks.class)
+@PostCondition(condition = AllChecks.class)
+public class BootstrapContextTestCase
+{
+   /** The user transaction */
+   @Resource(mappedName = "java:/eis/WorkConnectionFactory")
+   private WorkConnectionFactory wcf;
+
+   /** The user transaction */
+   @Resource(mappedName = "java:/eis/WorkConnectionFactory2")
+   private WorkConnectionFactory wcf2;
+
+   /**
+    * The Bootstrap context bean and friends
+    *
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 1)
+   private InputStreamDescriptor createBootstrapContext() throws Throwable
+   {
+      ClassLoader cl = Thread.currentThread().getContextClassLoader();
+      InputStreamDescriptor isd = new InputStreamDescriptor("second-bc.xml",
+            cl.getResourceAsStream("second-bc.xml"));
+      return isd;
+
+   }
+
+   /**
+    * The resource adapter
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 2)
+   private ResourceAdapterArchive createResourceAdapter() throws Throwable
+   {
+      return ResourceAdapterFactory.createWorkRar();
+   }
+   
+   /**
+    * The activation
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 3)
+   private ResourceAdaptersDescriptor createActivation() throws Throwable
+   {
+      return ResourceAdapterFactory.createWorkDeployment(null, "WorkConnectionFactory");
+   }
+
+   /**
+    * The activation
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 4)
+   private ResourceAdaptersDescriptor createActivation2() throws Throwable
+   {
+      return ResourceAdapterFactory.createWorkDeployment("Second", "WorkConnectionFactory2");
+   }
+   
+   /**
+    * Deployment
+    * @throws Throwable In case of an error
+    */
+   @Test
+   public void testDeployment() throws Throwable
+   {
+      assertNotNull(wcf);
+      assertNotNull(wcf2);
+
+      WorkConnection wc = wcf.getConnection();
+      assertNotNull(wc);
+      WorkConnection wc2 = wcf2.getConnection();
+      assertNotNull(wc2);
+
+      assertNotEquals(wc.getWorkManager(), wc2.getWorkManager());
+      assertEquals("Default", ((WorkManagerImpl) wc.getWorkManager()).getName());
+      assertEquals("WorkManager2", ((WorkManagerImpl) wc2.getWorkManager()).getName());
+
+      wc.close();
+   }
```

Comment:
Should also contain a test case that compares the BCs for the two deployments.

Or rename this test case to CustomWorkManagerTestCase, and add a new BootstrapContextTestCase based on test.rar, that exposes the various system identities from the BC